### PR TITLE
Renaming the IHDictionary and IHReplicatedDictionary 

### PR DESCRIPTION
**Pull request by asimarslan**
Renaming the IHDictionary and IHReplicatedDictionary  methods and introduce a shared interface IHDictionaryBase

### CODE REVIEW

**Comment by asimarslan** [comment]
run-linux

File: src/Hazelcast.Net/DistributedObjects/IHDictionary.Getting.cs [hunk 1]
```diff
@@ -21,98 +21,83 @@ namespace Hazelcast.DistributedObjects
...
+        /// <para>
+        /// The returned readonly dictionary is <b>NOT</b> backed by the hazelcast dictionary,
+        /// so changes to the returned readonly dictionary are <b>NOT</b> reflected in the <see cref="IHDictionary{TKey,TValue}"/>,
+        /// and vice-versa.
+        /// </para>
+        /// <para>
+        /// The <paramref name="predicate"/> must be serializable via Hazelcast serialization,
+        /// and have a counterpart on the server.</para>
+        /// </remarks>
+        Task<IReadOnlyDictionary<TKey, TValue>> GetEntriesAsync(IPredicate predicate);
```
Comment (zpqrtbnk) [inline src/Hazelcast.Net/DistributedObjects/IHDictionary.Getting.cs:93]: OK I am still not sure about this. `Get` gets a single thing, OK. `GetAll` gets everything, OK. But we are using `GetAll` with a collection of keys, and `GetEntries` with a predicate, that does not feel consistent. Thinking here...
* `Get` returns _one_ thing
* `GetAll` returns _everything_, unfiltered
* `GetMany` returns _many_ things, either from a list of keys, a predicate, whatever
* `GetStats` returns entry stats (and we rename `IHDictionaryEntry` into `IHDictionaryStats`

What do you think?

**Review by zpqrtbnk — COMMENTED** [review]
Have added comments for a few points we should discuss, I believe.

File: src/Hazelcast.Net/Data/IndexConfig.cs [hunk 2]
```diff
@@ -77,6 +77,19 @@ public IndexConfig AddAttribute(string attribute)
             return this;
         }
 
+        public IndexConfig AddAttributes(params string[] attributes)
```
[the remark this answers is not in the document]
Comment (asimarslan) [reply src/Hazelcast.Net/Data/IndexConfig.cs:85]: done

**Review by zpqrtbnk — APPROVED** [review]
ok

**Pull request merged**